panic: interface conversion: net.Addr is nil, not *net.UDPAddr

When one participant tries to share the screen, screego crashes with the following errors

```
2021-03-17T15:10:27Z DBG Logger initialized
2021-03-17T15:10:27Z INF SCREEGO_SECRET unset, user logins will be invalidated on restart
2021-03-17T15:10:27Z WRN Less than 40 ports are available for turn. When using multiple TURN connections this may not be enough
2021-03-17T15:10:27Z INF Users file not specified
2021-03-17T15:10:27Z DBG Using Port Range max=30005 min=30000
2021-03-17T15:10:27Z INF Start TURN/STUN addr=0.0.0.0:30000
2021-03-17T15:10:27Z INF Start HTTP addr=0.0.0.0:5050
2021-03-17T15:10:27Z DBG TURN username not found addr={"IP":"10.42.1.0","Port":62785,"Zone":""} username=c191no9fotf11rh6m340client
turn ERROR: 2021/03/17 15:10:27 error when handling datagram: failed to handle Allocate-request from 10.42.1.0:62785: no such user exists c191no9fotf11rh6m340client
2021-03-17T15:10:27Z DBG TURN username not found addr={"IP":"10.42.1.0","Port":62785,"Zone":""} username=c191no9fotf11rh6m340client
turn ERROR: 2021/03/17 15:10:27 error when handling datagram: failed to handle Allocate-request from 10.42.1.0:62785: no such user exists c191no9fotf11rh6m340client
2021-03-17T15:10:27Z DBG TURN username not found addr={"IP":"10.42.1.0","Port":62785,"Zone":""} username=c191no9fotf11rh6m340client
turn ERROR: 2021/03/17 15:10:27 error when handling datagram: failed to handle Allocate-request from 10.42.1.0:62785: no such user exists c191no9fotf11rh6m340client
2021-03-17T15:10:30Z DBG TURN username not found addr={"IP":"10.42.1.0","Port":27958,"Zone":""} username=c191no9fotf11rh6m340client
turn ERROR: 2021/03/17 15:10:30 error when handling datagram: failed to handle Allocate-request from 10.42.1.0:27958: no such user exists c191no9fotf11rh6m340client
2021-03-17T15:10:33Z DBG TURN username not found addr={"IP":"10.42.1.0","Port":27958,"Zone":""} username=c191no9fotf11rh6m340client
turn ERROR: 2021/03/17 15:10:33 error when handling datagram: failed to handle Allocate-request from 10.42.1.0:27958: no such user exists c191no9fotf11rh6m340client
2021-03-17T15:11:11Z DBG WebSocket New Connection id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:11Z DBG HTTP duration="496.573µs" host=ki.iznet ip=10.42.0.78:38854 path=/stream size=0 status=0
2021-03-17T15:11:11Z DBG WebSocket Receive event=*ws.Create id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:11Z DBG WebSocket Send event=room id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:24Z DBG HTTP duration="473.899µs" host=ki.iznet ip=10.42.0.78:38930 path=/ size=2158 status=200
2021-03-17T15:11:24Z DBG HTTP duration="389.456µs" host=ki.iznet ip=10.42.0.78:38930 path=/static/css/main.08e397cf.chunk.css size=0 status=200
2021-03-17T15:11:25Z DBG HTTP duration=294.806108ms host=ki.iznet ip=10.42.0.78:38936 path=/static/js/main.4606f178.chunk.js size=0 status=200
2021-03-17T15:11:25Z DBG HTTP duration=302.932205ms host=ki.iznet ip=10.42.0.78:38930 path=/static/js/2.6c008a11.chunk.js size=0 status=200
2021-03-17T15:11:25Z DBG HTTP duration="156.296µs" host=ki.iznet ip=10.42.0.78:38930 path=/config size=70 status=200
2021-03-17T15:11:25Z DBG WebSocket New Connection id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:25Z DBG HTTP duration="261.633µs" host=ki.iznet ip=10.42.0.78:38942 path=/stream size=0 status=0
2021-03-17T15:11:25Z DBG HTTP duration=78.182203ms host=ki.iznet ip=10.42.0.78:38930 path=/static/media/logo.d2ceaca5.svg size=0 status=200
2021-03-17T15:11:26Z DBG WebSocket Receive event=*ws.Join id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:26Z DBG WebSocket Send event=room id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:26Z DBG WebSocket Send event=room id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Receive event=*ws.StartShare id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=clientsession id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=hostsession id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=room id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=room id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Receive event=*ws.HostOffer id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=hostoffer id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Receive event=*ws.HostICE id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=hostice id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Receive event=*ws.HostICE id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Receive event=*ws.HostICE id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Receive event=*ws.HostICE id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Receive event=*ws.HostICE id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=hostice id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=hostice id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=hostice id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=hostice id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Receive event=*ws.HostICE id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=hostice id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Receive event=*ws.HostICE id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=hostice id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG TURN authenticated addr=10.42.1.0:48827 realm=screego
2021-03-17T15:11:37Z DBG TURN allocated addr=[::]:30005 relayaddr=10.20.20.40:30005
2021-03-17T15:11:37Z DBG WebSocket Receive event=*ws.HostICE id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Receive event=*ws.HostICE id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=hostice id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Receive event=*ws.HostICE id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Receive event=*ws.HostICE id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=hostice id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=hostice id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Receive event=*ws.HostICE id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=hostice id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=hostice id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Receive event=*ws.ClientAnswer id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=clientanswer id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Receive event=*ws.ClientICE id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=clientice id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Receive event=*ws.ClientICE id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=clientice id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG TURN authenticated addr=10.42.1.0:54097 realm=screego
2021-03-17T15:11:37Z DBG TURN allocated addr=[::]:30002 relayaddr=10.20.20.40:30002
2021-03-17T15:11:37Z DBG WebSocket Receive event=*ws.ClientICE id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Receive event=*ws.ClientICE id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Receive event=*ws.ClientICE id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Receive event=*ws.ClientICE id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=clientice id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=clientice id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=clientice id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG WebSocket Send event=clientice id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:37Z DBG TURN authenticated addr=10.42.1.0:3799 realm=screego
2021-03-17T15:11:37Z DBG TURN allocated addr=[::]:30003 relayaddr=10.20.20.40:30003
2021-03-17T15:11:37Z DBG TURN authenticated addr=10.42.1.0:1652 realm=screego
2021-03-17T15:11:37Z DBG TURN allocated addr=[::]:30004 relayaddr=10.20.20.40:30004
2021-03-17T15:11:38Z DBG WebSocket Receive event=*ws.HostICE id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:38Z DBG WebSocket Receive event=*ws.HostICE id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:38Z DBG WebSocket Send event=hostice id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:38Z DBG WebSocket Send event=hostice id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:38Z DBG TURN authenticated addr=10.42.1.0:4909 realm=screego
2021-03-17T15:11:38Z DBG TURN allocated addr=[::]:30001 relayaddr=10.20.20.40:30001
2021-03-17T15:11:38Z DBG WebSocket Receive event=*ws.ClientICE id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:38Z DBG WebSocket Send event=clientice id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:38Z DBG WebSocket Receive event=*ws.HostICE id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:38Z DBG WebSocket Send event=hostice id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:38Z DBG WebSocket Receive event=*ws.HostICE id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:38Z DBG WebSocket Send event=hostice id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:38Z DBG WebSocket Receive event=*ws.ClientICE id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:38Z DBG WebSocket Send event=clientice id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:38Z DBG WebSocket Receive event=*ws.HostICE id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:38Z DBG WebSocket Receive event=*ws.HostICE id=c191o79fotf0grs51p5g ip=10.42.1.240
2021-03-17T15:11:38Z DBG WebSocket Send event=hostice id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:38Z DBG WebSocket Send event=hostice id=c191o3pfotf0grs51p50 ip=10.42.1.240
2021-03-17T15:11:38Z DBG TURN authenticated addr=10.42.1.0:18791 realm=screego
panic: interface conversion: net.Addr is nil, not *net.UDPAddr

goroutine 24 [running]:
github.com/screego/server/turn.(*Generator).AllocatePacketConn(0xc0001428c0, 0xb20715, 0x4, 0x0, 0xc000605728, 0x40e0b8, 0xb0, 0xafeba0, 0xa61401, 0xc00027a580)
	/home/runner/work/server/server/turn/server.go:36 +0x3d8
github.com/pion/turn/v2/internal/allocation.(*Manager).CreateAllocation(0xc000334000, 0xc00041f410, 0xd85d60, 0xc00011e200, 0x0, 0x8bb2c97000, 0x2, 0x4, 0xc0002d6660)
	/home/runner/go/pkg/mod/github.com/pion/turn/v2@v2.0.5/internal/allocation/allocation_manager.go:95 +0x298
github.com/pion/turn/v2/internal/server.handleAllocateRequest(0xd85d60, 0xc00011e200, 0xd78960, 0xc00041f350, 0xc000336000, 0x94, 0x5dc, 0xc000334000, 0xc0002d6660, 0xc0002d29f0, ...)
	/home/runner/go/pkg/mod/github.com/pion/turn/v2@v2.0.5/internal/server/turn.go:115 +0xe1c
github.com/pion/turn/v2/internal/server.handleTURNPacket(0xd85d60, 0xc00011e200, 0xd78960, 0xc00041f350, 0xc000336000, 0x94, 0x5dc, 0xc000334000, 0xc0002d6660, 0xc0002d29f0, ...)
	/home/runner/go/pkg/mod/github.com/pion/turn/v2@v2.0.5/internal/server/server.go:72 +0x3e5
github.com/pion/turn/v2/internal/server.HandleRequest(0xd85d60, 0xc00011e200, 0xd78960, 0xc00041f350, 0xc000336000, 0x94, 0x5dc, 0xc000334000, 0xc0002d6660, 0xc0002d29f0, ...)
	/home/runner/go/pkg/mod/github.com/pion/turn/v2@v2.0.5/internal/server/server.go:42 +0x238
github.com/pion/turn/v2.(*Server).readLoop(0xc0001715e0, 0xd85d60, 0xc00011e200, 0xc000334000)
	/home/runner/go/pkg/mod/github.com/pion/turn/v2@v2.0.5/server.go:147 +0x1f8
github.com/pion/turn/v2.NewServer.func1(0xc0001715e0, 0xd85d60, 0xc00011e200, 0xd7d460, 0xc0001428c0)
	/home/runner/go/pkg/mod/github.com/pion/turn/v2@v2.0.5/server.go:74 +0x246
created by github.com/pion/turn/v2.NewServer
	/home/runner/go/pkg/mod/github.com/pion/turn/v2@v2.0.5/server.go:58 +0x21e
```

--- a/turn/portrange.go
+++ b/turn/portrange.go
@@ -11,7 +11,7 @@ import (
 type RelayAddressGeneratorPortRange struct {
 	MinPort uint16
 	MaxPort uint16
-	Rand randutil.MathRandomGenerator
+	Rand    randutil.MathRandomGenerator
 }
 
 func (r *RelayAddressGeneratorPortRange) Validate() error {
@@ -43,7 +43,7 @@ func (r *RelayAddressGeneratorPortRange) AllocatePacketConn(network string, requ
 		return conn, relayAddr, nil
 	}
 
-	return nil, nil, errors.New("max retries exceeded")
+	return nil, nil, errors.New("could not find free port: max retries exceeded")
 }
 
 func (r *RelayAddressGeneratorPortRange) AllocateConn(network string, requestedPort int) (net.Conn, net.Addr, error) {

--- a/turn/server.go
+++ b/turn/server.go
@@ -33,6 +33,9 @@ type Generator struct {
 
 func (r *Generator) AllocatePacketConn(network string, requestedPort int) (net.PacketConn, net.Addr, error) {
 	conn, addr, err := r.RelayAddressGenerator.AllocatePacketConn(network, requestedPort)
+	if err != nil {
+		return conn, addr, err
+	}
 	relayAddr := *addr.(*net.UDPAddr)
 	if r.ipv6 == nil || (relayAddr.IP.To4() != nil && r.ipv4 != nil) {
 		relayAddr.IP = r.ipv4

